### ISSUE TITLE
Support folder exludes

### ISSUE THREAD

**Issue by LouieK22** [Kampfkarren/selene#105]
I currently have some vendor libraries that I could care less about errors within. It would be nice to be able to exclude these directories from Selene so the errors don't show up in tools such a CI system.

I'm currently working around this by storing the libraries as .rbxmx files, but I would prefer to avoid this.

**Comment by Kampfkarren** [comment]
Duplicate of #15?

**Comment by LouieK22** [comment]
I wouldn't say it's a complete duplicate. I think a dedicated exclude property could be more straightforward instead of configuring a directory to not be linted.

**Comment by Kampfkarren** [comment]
If you don't want the errors to show up in a CI system, just don't pass it into selene? `selene src/` and such will ignore whatever vendor files you have, which shouldn't inseparable from your source code anyway.

**Comment by dejayc** [comment]
> ...will ignore whatever vendor files you have, which shouldn't inseparable from your source code anyway.

That's a pretty strong statement, to prescribe how everyone should structure their source code hierarchy, irrespective of mitigating circumstances.

**Comment by Kampfkarren** [comment]
Every large codebase I know structures their code this way. I don't think it's in anyone's best interest for Selene to support wacky codebases. Do you have a reasonably large example that does something different?

**Comment by dejayc** [comment]
> Every large codebase I know structures their code this way. I don't think it's in anyone's best interest for Selene to support wacky codebases. Do you have a reasonably large example that does something different?

I've been exposed to all sorts of code bases during the past 30 years, in every type of sector, company, level of developer maturity, etc.  Generally, tools that allow for flexibility in accommodating different scenarios are vastly more useful than single-scenario implementations.

That said, it would be nice if developers/DevOps roles/etc. could flexibly configure selene to behave the same regardless of whether selene was invoked from the command line, or VS Code extension, etc.

**Comment by 13k** [comment]
I agree with @dejayc. Assuming project layout sounds like an anti-feature. I think it's fine to have opinionated linting rules, but I don't think it's a linter's job to assume/enforce project structure.

Also, project layout is not always defined by the developer. Some platforms/projects are required to follow a given layout.

An example is neovim configuration using Lua. My (simplified) repo looks like this:

```
.
├── after # loaded late
│  └── some_plugin
│     └── late_plugin_config.lua
├── filetype.lua # loaded automatically by neovim
├── ftplugin # loaded automatically by file type
│  ├── ruby.lua
│  ├── rust.lua
│  └── another_filetype.lua
├── init.lua # config entry-point, but it's not loaded first
├── lua # added to path, available for `require()`
│  └── my_config
│     └── some_config_helpers.lua
├── pack
│  └── vendor
│     └── start
│        ├── Penlight
│        └── other_vendor_code
├── plugin # loaded early
│  └── some_early_config.lua
...
```

It's complicated to explain neovim's [startup procedure](https://neovim.io/doc/user/starting.html#startup) (which extends vim's), but basically you can have vendored code under `/after`, `/lua`, `/pack`, `/plugin` (probably somewhere else I'm missing) and there's no enforced layout or best practice because load order depends on the directory. You can also have authored code inside these same directories.

So it's a nightmare to pick and choose every loose file and directory that I'm authoring as opposed to just `selene .` with an ignore/exclude option to filter out vendored patterns.

Another example of project layout that is enforced by the platform is Dota 2 custom games. There's a single directory where you should place all the Lua files, including vendored code. This single directory is added to the Lua path, so everything under it must follow Lua path rules, meaning you can't split into `/src` and `/vendor` for example. You can package your own code into a separate sub-directory on your own package, but there are entry-point files that must go in the root, so `selene my-subdir` also doesn't work.

**Comment by 13k** [comment]
> Do you have a reasonably large example that does something different?

@Kampfkarren are you open to reconsider this issue? I've given a couple valid examples of project layouts that would benefit from such an option. neovim configs being extremely popular nowadays but, from what I see, they are lagging behind in adopting selene.

Another reason to consider it is that Lua doesn't enforce or recommend any given layout. Like Javascript or Go, the project can be structured pretty much in any way (again, sometimes it's even outside developers ability to define a layout). If it were like Rust, where there's a recommended layout, then it would make more sense to avoid configuration complexity.

Lastly, I don't see why project size matters here.

**Comment by Kampfkarren** [comment]
To be clear, I'm not against this, this was just closed for being a duplicate of #15.
